src/renderer/ipc.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable no-restricted-globals */
+/* eslint-disable no-console */
+import { logger } from '@sentry/utils';
+
+import { IPCChannel, IPCInterface, PROTOCOL_SCHEME } from '../common';
+
+/** Gets the available IPC implementation */
+function getImplementation(): IPCInterface {
+  // Favour IPC if it's been exposed by a preload script
+  if (window.__SENTRY_IPC__) {
+    return window.__SENTRY_IPC__;
+  }
+
+  logger.log('IPC was not configured in preload script, falling back to custom protocol and fetch');
+
+  fetch(`${PROTOCOL_SCHEME}://${IPCChannel.PING}`).catch(() =>
+    console.error(`Sentry SDK failed to establish connection with the Electron main process.
+ - Ensure you have initialized the SDK in the main process
+ - If your renderers use custom sessions, be sure to set 'getSessions' in the main process options
+ - If you are bundling your main process code and using Electron < v5, you'll need to manually configure a preload script`),
+  );
+
+  // We include sentry_key in the URL so these dont end up in fetch breadcrumbs
+  // https://github.com/getsentry/sentry-javascript/blob/a3f70d8869121183bec037571a3ee78efaf26b0b/packages/browser/src/integrations/breadcrumbs.ts#L240
```

Comment:
This is kinda hacky - but a nice solution

Reply:
Yeah, maybe a bit fragile. I initially attempted to filter the breadcrumbs when they're sent from renderer to main but realised that scope updates were triggering fetch, which was triggering scope updates, which.... yeah not great!

Might be nice to have a magic way to disable Sentry fetch breadcrumbs via a magic header?

Reply:
Lemme ask around to see what others think.

Reply:
We're pinned to a specific version of `@sentry/browser` so this is only prone to breaking on dependency updates. Shall we merge this as it is and I'll open an issue on `getsentry/javascript` to track this?

Reply:
Yes please, opening an issue would be great